app/encryption.py

```diff
@@ -28,20 +28,34 @@ class NotificationDictToSign(TypedDict):
 
 
 class CryptoSigner:
-    def init_app(self, app: Any, secret_key: str | Iterator[str], salt: str) -> None:
+    def init_app(self, app: Any, secret_key: str | List[str], salt: str) -> None:
+        self.app = app
+        self.secret_key = cast(List[str], [secret_key] if type(secret_key) is str else secret_key)
         self.serializer = URLSafeSerializer(secret_key)
         self.salt = salt
         self.dangerous_salt = app.config.get("DANGEROUS_SALT")
 
     def sign(self, to_sign: str | NotificationDictToSign) -> str | bytes:
         return self.serializer.dumps(to_sign, salt=self.salt)
 
+    def sign_with_all_keys(self, to_sign: str | NotificationDictToSign) -> List[str | bytes]:
+        signed = []
+        for k in reversed(self.secret_key):  # reversed so that the default key is last
+            signed.append(URLSafeSerializer(k).dumps(to_sign, salt=self.salt))
+        return signed
+
     # TODO: get rid of this after everything is signed with the new salts
     # This is only needed where we look things up by the signed value:
     #     - get_api_key_by_secret()
     def sign_dangerous(self, to_sign: str) -> str | bytes:
         return self.serializer.dumps(to_sign, salt=self.dangerous_salt)
 
+    def sign_dangerous_with_all_keys(self, to_sign: str | NotificationDictToSign) -> List[str | bytes]:
```

Comment:
I'd rename that one to mean that we are signing with the old salt, as saying sign dangerous is not obvious on what is happening.